conmon-rs/server/src/listener.rs

```diff
@@ -0,0 +1,32 @@
+use anyhow::{bail, Context, Result};
+use std::{
+    fs::File,
+    os::unix::io::AsRawFd,
+    path::{Path, PathBuf},
+};
+use tokio::net::UnixListener;
+
+pub fn bind_long_path(path: &Path) -> Result<UnixListener> {
+    let parent = match path.parent() {
+        Some(p) => p,
+        None => bail!(
+            "Tried to specify / as socket to bind to: {}",
+            path.display()
+        ),
+    };
```

Comment:
nit, you can use the Context on Option types too:
```suggestion
    let parent = path.parent().context(format!(
        "tried to specify / as socket to bind to: {}",
        path.display()
    ))?;
```

Reply:
See https://github.com/containers/conmon-rs/pull/312